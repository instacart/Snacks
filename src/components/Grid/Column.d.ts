import * as React from 'react'
import { RadiumStyles } from '../..'
import { Size } from '../../styles/responsive'

export interface ColumnProps {
  children?: React.ReactNode

  /** object where keys are breakpoint and value is number of columns to span at that breakpoint */
  sizes?: { [Key in Size]?: number }

  style?: RadiumStyles
}

declare const Column: React.ComponentType<ColumnProps>

export default Column
