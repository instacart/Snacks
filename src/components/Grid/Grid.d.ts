import * as React from 'react'
import { RadiumStyles } from '../..'

export interface GridProps {
  children?: React.ReactNode
  style?: RadiumStyles
}

declare const Grid: React.ComponentType<GridProps>

export default Grid
