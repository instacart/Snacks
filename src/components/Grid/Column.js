import React from 'react'
import PropTypes from 'prop-types'
import responsive from '../../styles/responsive'
import { supportsCSSGrid } from '../../utils/detectFeature'

const { columnWidth } = responsive

const getLecacySizedStyles = sizes => {
  return {
    display: 'inline-block',

    [responsive.xs]: {
      display: 'block',
      width: '100%',
    },

    [responsive.sm]: {
      width: `${(sizes.sm || 1) * columnWidth}px`,
    },

    [responsive.md]: {
      width: `${(sizes.md || sizes.sm || 1) * columnWidth}px`,
    },

    [responsive.mdLg]: {
      width: `${(sizes.mdLg || sizes.md || sizes.sm || 1) * columnWidth}px`,
    },

    [responsive.lg]: {
      width: `${(sizes.lg || sizes.mdLg || sizes.md || sizes.sm || 1) * columnWidth}px`,
    },

    [responsive.xl]: {
      width: `${(sizes.xl || sizes.lg || sizes.mdLg || sizes.md || sizes.sm || 1) * columnWidth}px`,
    },
  }
}

const getSizedStyles = sizes => {
  return {
    [responsive.sm]: {
      gridColumn: `span ${sizes.sm || 1}`,
    },

    [responsive.md]: {
      gridColumn: `span ${sizes.md || sizes.sm || 1}`,
    },

    [responsive.mdLg]: {
      gridColumn: `span ${sizes.mdLg || sizes.md || sizes.sm || 1}`,
    },

    [responsive.lg]: {
      gridColumn: `span ${sizes.lg || sizes.mdLg || sizes.md || sizes.sm || 1}`,
    },

    [responsive.xl]: {
      gridColumn: `span ${sizes.xl || sizes.lg || sizes.mdLg || sizes.md || sizes.sm || 1}`,
    },
  }
}

const Column = props => {
  const { className, sizes } = props

  if (props.sizes.xs) {
    console.warn(
      'xs size prop passed to Column!',
      'This will be ignored. All columns at xs screen size are full-width. ',
      'Please remove this definition. Sizes passed: ',
      props.sizes
    )
  }

  const styles = supportsCSSGrid() ? getSizedStyles(sizes) : getLecacySizedStyles(sizes)

  return (
    <div className={className} css={[styles, props.style]}>
      {props.children}
    </div>
  )
}

Column.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  /** object where keys are breakpoint and value is number of columns to span at that breakpoint */
  sizes: PropTypes.shape({
    xs: PropTypes.number, // should never be passed
    sm: PropTypes.number,
    md: PropTypes.number,
    mdLg: PropTypes.number,
    lg: PropTypes.number,
    xl: PropTypes.number,
  }),
  /**
   * Optional style overrides merged into emotion css
   *
   * @deprecated
   * This prop exists for backwards compatibility and will be
   * removed in a future version
   */
  style: PropTypes.object,
}

Column.defaultProps = {
  sizes: {},
}

export default Column
