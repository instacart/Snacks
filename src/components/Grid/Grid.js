import React from 'react'
import PropTypes from 'prop-types'
import Radium from 'radium'
import responsive from '../../styles/responsive'

const styles = {
  marginTop: 0,
  marginRight: 'auto',
  marginBottom: 0,
  marginLeft: 'auto',

  [responsive.xs]: {
    width: '100%',
  },

  [responsive.sm]: {
    width: responsive.screenWidths.sm,
  },

  [responsive.md]: {
    width: responsive.screenWidths.md,
  },

  [responsive.mdLg]: {
    width: responsive.screenWidths.mdLg,
  },

  [responsive.lg]: {
    width: responsive.screenWidths.lg,
  },

  [responsive.xl]: {
    width: responsive.screenWidths.xl,
  },
}

const Grid = props => {
  return <div style={[styles, props.style]}>{props.children}</div>
}

Grid.propTypes = {
  children: PropTypes.node,
  /** Optional style overrides */
  style: PropTypes.object,
}

export default Radium(Grid)
