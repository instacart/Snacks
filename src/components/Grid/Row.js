import React      from 'react'
import PropTypes  from 'prop-types'
import Radium from 'radium'
import responsive from '../../styles/responsive'
import { supportsCSSGrid } from '../../utils/detectFeature'

// used for browsers without css grid support
const legacyStyles = {
  display: 'block',

  [responsive.sm]: {
    width: `${6 * responsive.columnWidth}px`
  },

  [responsive.md]: {
    width: `${8 * responsive.columnWidth}px`
  },

  [responsive.mdLg]: {
    width: `${10 * responsive.columnWidth}px`
  },

  [responsive.lg]: {
    width: `${12 * responsive.columnWidth}px`
  },

  [responsive.xl]: {
    width: `${14 * responsive.columnWidth}px`
  }
}

const styles = {
  [responsive.xs]: {
    display: 'block'
  },

  [responsive.sm]: {
    display: 'grid',
    gridTemplateColumns: `repeat(6, ${responsive.columnWidth}px)`
  },

  [responsive.md]: {
    display: 'grid',
    gridTemplateColumns: `repeat(8, ${responsive.columnWidth}px)`
  },

  [responsive.mdLg]: {
    display: 'grid',
    gridTemplateColumns: `repeat(10, ${responsive.columnWidth}px)`
  },

  [responsive.lg]: {
    display: 'grid',
    gridTemplateColumns: `repeat(12, ${responsive.columnWidth}px)`
  },

  [responsive.xl]: {
    display: 'grid',
    gridTemplateColumns: `repeat(14, ${responsive.columnWidth}px)`
  }
}

const getMaxColumnsStyles = props => {
  const { maxColumns } = props

  const override = supportsCSSGrid() ? {
    gridTemplateColumns: `repeat(${maxColumns}, ${responsive.columnWidth}px)`,
    justifyContent: 'center'
  } : {
    width: `${maxColumns * responsive.columnWidth}px`,
    marginTop: 0,
    marginRight: 'auto',
    marginBottom: 0,
    marginLeft: 'auto',
  }

  if (maxColumns <= 6) {
    return {
      [responsive.md]: override,
      [responsive.mdLg]: override,
      [responsive.lg]: override,
      [responsive.xl]: override
    }
  } else if (maxColumns <= 8) {
    return {
      [responsive.mdLg]: override,
      [responsive.lg]: override,
      [responsive.xl]: override
    }
  } else if (maxColumns <= 10) {
    return {
      [responsive.lg]:override,
      [responsive.xl]: override
    }
  } else if (maxColumns <= 12) {
    return {
      [responsive.xl]: override
    }
  }

  return {}
}

const fullWidthStyles = ['xs', 'sm', 'md', 'mdLg', 'lg', 'xl'].reduce(
  (styles, size) => ({
    ...styles,
    [responsive[size]]: {
      display: 'block',
      marginTop: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`,
      marginRight: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`,
      marginBottom: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`,
      marginLeft: `0 calc(-1 * (100vw - ${responsive.screenWidths[size]}px) / 2)`
    }
  }),
  { width: '100vw', justifyContent: 'center' }
)

const getFullWidthStyles = props => props.forceFullPage ? fullWidthStyles : {}

const Row = props => {
  const componentStyles = supportsCSSGrid() ? styles : legacyStyles

  return (
    <div
      style={[
        componentStyles,
        getMaxColumnsStyles(props),
        getFullWidthStyles(props),
        props.style
      ]}
    >
      { props.children }
    </div>
  )
}

Row.propTypes = {
  /** Force Row to width of 100vw -- Snacks will add negative margin */
  forceFullPage: PropTypes.bool,
  /** Maximum number of columns this Row should grow to as screen width increases. Cannot exceed 14. */
  maxColumns: PropTypes.number,
  /** Optional style overrides */
  style: PropTypes.object
}

Row.defaultProps = {
  maxColumns: 14
}

export default Radium(Row)
