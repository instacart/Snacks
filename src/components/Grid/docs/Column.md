Column Example without a parent Grid or Row:

    columnStyles = { backgroundColor: '#43B02A', border: '1px solid white', textAlign: 'center' };
    <div>
      <Column sizes={{ sm: 1 }} styles={columnStyles}><p>One</p></Column>
      <Column sizes={{ md: 3 }} styles={columnStyles}><p>Two</p></Column>
      <Column sizes={{ sm: 2, md: 1, mdLg: 4, lg: 5, xl: 4 }} styles={columnStyles}><p>Three</p></Column>
    </div>
