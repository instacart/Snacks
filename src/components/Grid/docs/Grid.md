Grid example:

    columnStyles = { backgroundColor: '#43B02A', border: '1px solid white', textAlign: 'center' };
    <div>
      <Grid styles={{ backgroundColor: '#eee' }}>
        <Row styles={{ marginBottom: '20px' }}>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Seven</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Eight</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Nine</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Ten</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Eleven</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Twelve</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Thirteen</p></Column>
          <Column sizes={{ sm: 1 }} styles={columnStyles}><p>Fourteen</p></Column>
        </Row>
        <Row styles={{ marginBottom: '20px' }}>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Seven</p></Column>
        </Row>
        <Row styles={{ marginBottom: '20px' }}>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Four</p></Column>
        </Row>
        <Row styles={{ marginBottom: '20px' }}>
          <Column sizes={{ sm: 4 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 4 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 4 }} styles={columnStyles}><p>Two</p></Column>
        </Row>
        <Row styles={{ marginBottom: '20px' }}>
          <Column sizes={{ sm: 2, md: 2, lg: 4, xl: 5}} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Four</p></Column>
        </Row>
        <Row styles={{ marginBottom: '20px' }} maxColumns={10}>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Six</p></Column>
        </Row>
        <Row styles={{ marginBottom: '20px' }} maxColumns={6}>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 3 }} styles={columnStyles}><p>Six</p></Column>
        </Row>
      </Grid>
    </div>
