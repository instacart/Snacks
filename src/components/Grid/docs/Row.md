Grid example:

    rowStyles = { marginBottom: '20px' };
    columnStyles = { backgroundColor: '#43B02A', border: '1px solid white', textAlign: 'center' };
    <div>
      <Grid styles={{ backgroundColor: '#eee' }}>
        <Row styles={rowStyles} maxColumns={10}>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Seven</p></Column>
        </Row>
        <Row styles={rowStyles} maxColumns={6}>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>Seven</p></Column>
        </Row>
        <Row styles={rowStyles} forceFullPage={true}>
          <Column sizes={{ sm: 2 }} styles={columnStyles}><p>One</p></Column>
          <Column sizes={{ sm: 4 }} styles={columnStyles}><p>Two</p></Column>
          <Column sizes={{ sm: 6 }} styles={columnStyles}><p>Three</p></Column>
          <Column sizes={{ sm: 8 }} styles={columnStyles}><p>Four</p></Column>
          <Column sizes={{ sm: 10 }} styles={columnStyles}><p>Five</p></Column>
          <Column sizes={{ sm: 12 }} styles={columnStyles}><p>Six</p></Column>
          <Column sizes={{ sm: 14 }} styles={columnStyles}><p>Seven</p></Column>
        </Row>
      </Grid>
    </div>
