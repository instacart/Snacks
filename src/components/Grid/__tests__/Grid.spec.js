import React from 'react'
import renderer from 'react-test-renderer'
import { StyleRoot } from 'radium'
import Grid from '../Grid'
import Row from '../Row'
import Column from '../Column'

it('renders Grid correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Grid />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Grid with styles correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Grid style={{ width: '100%', color: '#ccc' }} />
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Grid with Row correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Grid>
            <Row maxColumns={10} />
          </Grid>
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Grid with Row with Columns correctly', () => {
  const tree = renderer
    .create(
      <StyleRoot>
        <div>
          <Grid>
            <Row maxColumns={10}>
              <Column sizes={{ sm: 2, lg: 4, xl: 5 }} />
              <Column sizes={{ sm: 2, lg: 4, xl: 5 }} />
              <Column sizes={{ sm: 2, lg: 4, xl: 5 }} />
            </Row>
          </Grid>
        </div>
      </StyleRoot>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
