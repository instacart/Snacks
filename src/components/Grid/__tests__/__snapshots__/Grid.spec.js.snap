// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Grid correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      className="rmq-2c87bfe9 rmq-136ec809 rmq-816a7dd1 rmq-b3aaa63a rmq-2e8763ff rmq-7dcad03d"
      data-radium={true}
      style={
        Object {
          "margin": "0 auto",
        }
      }
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-2c87bfe9{width: 100% !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-136ec809{width: 624px !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-816a7dd1{width: 832px !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-b3aaa63a{width: 1040px !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2e8763ff{width: 1248px !important;}}
@media (min-width: 1456px){ .rmq-7dcad03d{width: 1456px !important;}}",
      }
    }
  />
</div>
`;

exports[`renders Grid with Row correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      className="rmq-2c87bfe9 rmq-136ec809 rmq-816a7dd1 rmq-b3aaa63a rmq-2e8763ff rmq-7dcad03d"
      data-radium={true}
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      <div
        className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-820d5078 rmq-98718fd3"
        data-radium={true}
        style={Object {}}
      />
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-2c87bfe9{width: 100% !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-136ec809{width: 624px !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-816a7dd1{width: 832px !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-b3aaa63a{width: 1040px !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2e8763ff{width: 1248px !important;}}
@media (min-width: 1456px){ .rmq-7dcad03d{width: 1456px !important;}}
@media (max-width: 767px){ .rmq-e9fd10a8{display: block !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-94de0125{display: grid !important;
grid-template-columns: repeat(6, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-eeb4761a{display: grid !important;
grid-template-columns: repeat(8, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-3e10bf34{display: grid !important;
grid-template-columns: repeat(10, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2295bbb9{display: grid !important;
grid-template-columns: repeat(12, 104px) !important;}}
@media (min-width: 1456px){ .rmq-108b114{display: grid !important;
grid-template-columns: repeat(14, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-820d5078{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-98718fd3{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}",
      }
    }
  />
</div>
`;

exports[`renders Grid with Row with Columns correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      className="rmq-2c87bfe9 rmq-136ec809 rmq-816a7dd1 rmq-b3aaa63a rmq-2e8763ff rmq-7dcad03d"
      data-radium={true}
      style={
        Object {
          "margin": "0 auto",
        }
      }
    >
      <div
        className="rmq-e9fd10a8 rmq-94de0125 rmq-eeb4761a rmq-3e10bf34 rmq-2295bbb9 rmq-108b114 rmq-820d5078 rmq-98718fd3"
        data-radium={true}
        style={Object {}}
      >
        <div
          className="rmq-8f6274a rmq-9b0e1a5b rmq-a542a9ac rmq-f1cb9e65 rmq-dd3ac64f"
          data-radium={true}
          style={Object {}}
        />
        <div
          className="rmq-8f6274a rmq-9b0e1a5b rmq-a542a9ac rmq-f1cb9e65 rmq-dd3ac64f"
          data-radium={true}
          style={Object {}}
        />
        <div
          className="rmq-8f6274a rmq-9b0e1a5b rmq-a542a9ac rmq-f1cb9e65 rmq-dd3ac64f"
          data-radium={true}
          style={Object {}}
        />
      </div>
    </div>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-2c87bfe9{width: 100% !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-136ec809{width: 624px !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-816a7dd1{width: 832px !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-b3aaa63a{width: 1040px !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2e8763ff{width: 1248px !important;}}
@media (min-width: 1456px){ .rmq-7dcad03d{width: 1456px !important;}}
@media (max-width: 767px){ .rmq-e9fd10a8{display: block !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-94de0125{display: grid !important;
grid-template-columns: repeat(6, 104px) !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-eeb4761a{display: grid !important;
grid-template-columns: repeat(8, 104px) !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-3e10bf34{display: grid !important;
grid-template-columns: repeat(10, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2295bbb9{display: grid !important;
grid-template-columns: repeat(12, 104px) !important;}}
@media (min-width: 1456px){ .rmq-108b114{display: grid !important;
grid-template-columns: repeat(14, 104px) !important;}}
@media (min-width: 1248px) and (max-width: 1455px) { .rmq-820d5078{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}
@media (min-width: 1456px) { .rmq-98718fd3{grid-template-columns: repeat(10, 104px) !important;
justify-content: center !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-8f6274a{grid-column: span 2 !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-9b0e1a5b{grid-column: span 2 !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-a542a9ac{grid-column: span 2 !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-f1cb9e65{grid-column: span 4 !important;}}
@media (min-width: 1456px){ .rmq-dd3ac64f{grid-column: span 5 !important;}}",
      }
    }
  />
</div>
`;

exports[`renders Grid with styles correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <div
      className="rmq-2c87bfe9 rmq-136ec809 rmq-816a7dd1 rmq-b3aaa63a rmq-2e8763ff rmq-7dcad03d"
      data-radium={true}
      style={
        Object {
          "color": "#ccc",
          "margin": "0 auto",
          "width": "100%",
        }
      }
    />
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "@media (max-width: 767px){ .rmq-2c87bfe9{width: 100% !important;}}
@media (min-width: 768px) and (max-width: 831px){ .rmq-136ec809{width: 624px !important;}}
@media (min-width: 832px) and (max-width: 1039px){ .rmq-816a7dd1{width: 832px !important;}}
@media (min-width: 1040px) and (max-width: 1247px){ .rmq-b3aaa63a{width: 1040px !important;}}
@media (min-width: 1248px) and (max-width: 1455px){ .rmq-2e8763ff{width: 1248px !important;}}
@media (min-width: 1456px){ .rmq-7dcad03d{width: 1456px !important;}}",
      }
    }
  />
</div>
`;
